packages/components/src/input-control/stories/index.story.tsx

```diff
@@ -82,3 +89,32 @@ WithEdgeLabel.args = {
 	__unstableInputWidth: '20em',
 	labelPosition: 'edge',
 };
+
+export const ShowPassword: StoryFn< typeof InputControl > = ( args ) => {
+	const [ visible, toggleVisible ] = useReducer(
+		( value ) => ! value,
+		false
+	);
```

Comment:
I'm totally fine with this in our own code, but since this is a demo snippet for the public, I feel like we could keep it more approachable by sticking with `useState()`.